wizard.py

```diff
@@ -0,0 +1,1038 @@
+      decorator). Most views correspond to a page the user surfs to and
+      therefor render and return the according template on a GET request.
+      If a page has form postable content, the according view also accepts
+      a POST request which is usually sent when the user clicks NEXT as s/he
+      walks through the pages. The view then handles and persists the posted
+      data and in case everything went well redirects to the text view (page).
+      The views in this document are ordered from top to bottom in the same
+      order as the user walks from page to page.
+      Some pages have additional views associated that accept ajax calls,
+      e.g. to upload or remove files.
+
+"""
+import os
+import random
+import hashlib
+import time
+import StringIO
+import tarfile
+
+from functools import wraps
+from flask import (Flask, render_template, session, redirect, url_for, request,
+    flash, send_file, abort, json, jsonify, get_flashed_messages)
+from flask_pymongo import PyMongo
+from flask_wtf.csrf import CSRFProtect
+
+import in_toto.util
+import in_toto.models.link
+import in_toto.models.mock_link
+import in_toto.models.layout
+import in_toto.artifact_rules
+import securesystemslib.keys
+
+import tooldb
+import reverse_layout
+
+app = Flask(__name__, static_url_path="", instance_relative_config=True)
+mongo = PyMongo(app)
+csrf = CSRFProtect(app)
+
+app.config.update(dict(
+    DEBUG=True,
+    SECRET_KEY="do not use the development key in production!!!",
+))
+
+# Supply a config file at "instance/config.py" that carries
+# e.g. your deployment secret key
+app.config.from_pyfile("config.py")
+
+# Reload if a template has changed (only for development, i.e. in DEBUG mode)
+app.jinja_env.auto_reload = app.config["DEBUG"]
+
+
+# -----------------------------------------------------------------------------
+# Utils
+# -----------------------------------------------------------------------------
+def session_to_ssc(session_data):
+  """
+  <Purpose>
+    Takes a data from a user session, i.e. form posted data from multiple
+    pages (vcs, building, qa, ...) to generate a dictionary of lists of step
+    data and inspection data, i.e. a software supply chain (ssc).
+
+  <Returns>
+    Software Supply Chain data, i.e. a dictionary of step and inspection data
+    Note: these are not not actual in-toto Step and Inspection objects
+
+    e.g.:
+    {
+      steps: [
+        {
+          "name": <unique step name>,
+          "cmd": <expected command>,
+          "modifies": <boolean>  # whether the step modifies artifacts or not
+        }, ...
+      ],
+      inspections: [
+        {
+          "name": <unique inspection name>,
+          "cmd": <command to run inspection>,
+          "based_on": <step name whose products are used for that inspection>
+        }
+      ]
+    }
+  """
+  ssc_steps = []
+  ssc_inspections = []
+
+  for step_type in ["vcs", "building", "qa", "package"]:
```

Comment:
is this constant? you probably want to keep a data model of it or so....

Reply:
Good idea to tie down the data model! Currently it is rather relaxed. I receive form posted data, dump it to my [*key-value store and then write a distributed map reduce ...*](http://howfuckedismydatabase.com/nosql/).
Just kidding, I'll add a proper data model!

Reply:
See https://github.com/in-toto/layout-web-tool/issues/33